### PR TITLE
Add full API parity for search

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

File: boxsdk/object/search.py [hunk 1]
```diff
@@ -207,6 +227,30 @@ def search(
             Which content types to search. Valid types include name, description, file_content, comments, and tags.
         :type content_types:
             `Iterable` of `unicode`
+        :param scope:
+            The scope of content to search over
+        :type scope:
+            `unicode` or None
```
Comment (Jeff-Meadows) [inline boxsdk/object/search.py]:
```suggestion
            :class:`SearchScope`
```

File: boxsdk/object/search.py [hunk 2]
```diff
@@ -207,6 +227,30 @@ def search(
...
+        :type size_range:
+            (`int` or None, `int` or None)
+        :param owner_users:
+            Owner users to filter content by; only content belonging to these users will be returned.
+        :type owner_users:
+            `iterable` of :class:`User`
+        :param trash_content:
+            Whether to search trashed or non-trashed content.
+        :type trash_content:
+            `unicode` or None
```
Comment (Jeff-Meadows) [inline boxsdk/object/search.py]:
```suggestion
            :class:`TrashContent`
```

File: boxsdk/object/search.py [hunk 3]
```diff
@@ -207,6 +227,30 @@ def search(
...
+        :type updated_at_range:
+            (`unicode` or None, `unicode` or None)
+        :param size_range:
+            A tuple of the form (lower_bound, upper_bound) for the size in bytes of items to search.
+        :type size_range:
+            (`int` or None, `int` or None)
+        :param owner_users:
+            Owner users to filter content by; only content belonging to these users will be returned.
+        :type owner_users:
+            `iterable` of :class:`User`
```
Comment (Jeff-Meadows) [inline boxsdk/object/search.py]:
```suggestion
            `Iterable` of :class:`User`
```

File: HISTORY.rst [hunk 4]
```diff
@@ -10,6 +10,8 @@ Release History
 
 - Python 2.6 is no longer supported.
 - Python 3.3 is no longer supported.
+- ``client.search()`` now returns a ``Search`` object that exposes a ``query()`` method to call the Search API.
```
Comment (Jeff-Meadows) [inline HISTORY.rst]: What's the reasoning behind this change? Do we think the `Search` object may have more functionality in the future?

**Pull request merged**